### PR TITLE
Add Terraform Documentation for k8s, k8s-worker charms

### PR DESCRIPTION
**Pull request by louiseschmidtgen**
## Add Terraform docs for k8s, k8s-worker charms
This PR adds documentation on the k8s and k8s-worker terraform modules.

### CODE REVIEW

File: docs/src/charm/howto/install-terraform.md [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+## Authentication
+
+As a first step, authenticate the Terraform Juju Provider with the juju
+controller by choosing one of the options outlined in the
+[provider documentation][auth].
+
+## Terraform Module creation
+
+The Terraform deployment is done using a high-level module that specifies the
+juju model to deploy the submodules to. The k8s charm and the k8s-worker charm
```
Comment (nhennigan) [inline docs/src/charm/howto/install-terraform.md:23]: Nit: Juju model

File: docs/src/charm/howto/install-terraform.md [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+each live in a separate module referenced by the high-level module.
```
Comment (nhennigan) [inline docs/src/charm/howto/install-terraform.md:24]: Also a small change:
The high-level module also references the k8s charm and the k8s-worker charm that each live in a separate module.

**Review by nhennigan — COMMENTED** [review]
A few small nits to help improve the flow but overall great work :smiley_cat:

File: docs/src/charm/howto/install-terraform.md [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+### Charm modules
+
+The charm modules for the k8s and k8s-worker charms offer the following
+configuration options:
+
+| Name | Type | Description | Required | Default |
+| - | - | - | - | - |
+| `app_name`| string | Application name | False | k8s |
+| `base` | string | Ubuntu base to deploy the charm onto | False | ubuntu@24.04 |
+| `channel`| string | Channel that the charm is deployed from | False | 1.30/edge |
```
[the remark this answers is not in the document]
Comment (louiseschmidtgen) [reply docs/src/charm/howto/install-terraform.md:73]: I didn't catch this in the k8s-operator PR, so I'll bump it with the next terraform work (openstack) and its complementary documentation.

File: docs/src/charm/howto/install-terraform.md [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+
+```
+git clone https://github.com/canonical/k8s-operator.git
+```
+
+Find the control-plane module at `k8s-operator/charms/worker/k8s/terraform` and
+the k8s-worker module at `k8s-operator/tree/main/charms/worker/terraform`.
+
+```{note} 
+Please ensure that the high-level directory is pointing to the correct path of
```
[the remark this answers is not in the document]
Comment (louiseschmidtgen) [reply docs/src/charm/howto/install-terraform.md:99]: Great point, I checked the Terraform lingo: https://developer.hashicorp.com/terraform/docs/glossary#root-module.
We should be calling it a root and child module. I'll make the adjustments.

**Review by nhennigan — APPROVED** [review]
Thanks for applying those changes. :smiley:

**Pull request merged**